### ISSUE TITLE
[BUG] MPS backend out of memory for Timeseries tests

### ISSUE THREAD

**Issue by suzhoum** [autogluon/autogluon#4544]
**Bug Report Checklist**

- [ ] I provided code that demonstrates a minimal reproducible example.  
- [x] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [ ] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**

Multiple timeseries tests errored out when running on on MacOS while passed on both Linux and Windows. It seems that it was trying to fit with the GPU mode on MPS but the support was not there yet or needs to be configured. 

**Expected behavior**

**To Reproduce**

Run platform tests on AutoGluon CI

**Screenshots / Logs**

Platform tests were run on:
https://github.com/suzhoum/autogluon/actions/runs/11356061304/job/31591048836

Sample log from the failure:
```
self = <[RuntimeError('MPS backend out of memory (MPS allocated: 528.00 MB, other allocations: 1.20 MB, max allowed: 7.93 GB)...isable upper limit for memory allocations (may cause system failure).') raised in repr()] Tensor object at 0x310691db0>
other = 1.0

    @_handle_torch_function_and_wrap_type_error_to_not_implemented
    def __rsub__(self, other):
>       return _C._VariableFunctions.rsub(self, other)
E       RuntimeError: MPS backend out of memory (MPS allocated: 528.00 MB, other allocations: 1.20 MB, max allowed: 7.93 GB). Tried to allocate 256 bytes on shared pool. Use PYTORCH_MPS_HIGH_WATERMARK_RATIO=0.0 to disable upper limit for memory allocations (may cause system failure).

../../../../miniconda3/envs/autogluon_py3/lib/python3.9/site-packages/torch/_tensor.py:962: RuntimeError
_ ERROR at setup of test_when_fit_called_then_models_train_and_all_scores_can_be_computed[MASE-1-PatchTSTModel] _

dummy_hyperparameters = {'epochs': 1, 'model_path': 'autogluon/chronos-t5-tiny', 'n_jobs': 1, 'num_batches_per_epoch': 1, ...}

    @pytest.fixture(scope="module")
    def trained_models(dummy_hyperparameters):
        models = {}
        model_paths = []
        for model_class, prediction_length in itertools.product(TESTABLE_MODELS, TESTABLE_PREDICTION_LENGTHS):
            temp_model_path = tempfile.mkdtemp()
            model = model_class(
                path=temp_model_path,
                freq="h",
                prediction_length=prediction_length,
                hyperparameters=dummy_hyperparameters,
            )
    
            model.fit(train_data=DUMMY_TS_DATAFRAME)
>           model.score_and_cache_oof(DUMMY_TS_DATAFRAME, store_val_score=True, store_predict_time=True)

tests/unittests/models/test_models.py:72: 


FAILED tests/unittests/models/test_chronos.py::test_when_context_length_not_provided_then_context_length_set_to_dataset_length[5-5] - RuntimeError: MPS backend out of memory (MPS allocated: 64.00 MB, other allocations: 1.06 MB, max allowed: 7.93 GB). Tried to allocate 256 bytes on shared pool. Use PYTORCH_MPS_HIGH_WATERMARK_RATIO=0.0 to disable upper limit for memory allocations (may cause system failure).
```

**Installed Versions**

python3.10

</details>

**Comment by canerturkmen** [comment]
I was not able to reproduce, although I got a different error on the MPS backend for torch 2.3. #4565 disables MPS.

**Comment by shchur** [comment]
Should be fixed by #4565
